tasmota/xnrg_12_solaxX1.ino

```diff
@@ -104,295 +101,364 @@ struct SOLAXX1 {
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: wPowerfactor_data: %s"), TempDataChar);
+        dtostrfd(DataRead[37] * 0.01f, 2, TempDataChar);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: wUpperLimit: %s (overexcite limits)"), TempDataChar);
+        dtostrfd(DataRead[38] * 0.01f, 2, TempDataChar);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: wLowerLimit: %s (underexcite limits)"), TempDataChar);
+        dtostrfd(DataRead[39] * 0.01f, 2, TempDataChar);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: wPowerLow: %s (power ratio change upper limits)"), TempDataChar);
+        dtostrfd(DataRead[40] * 0.01f, 2, TempDataChar);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: wPowerUp: %s (power ratio change lower limits)"), TempDataChar);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: Qpower_set: %d"), (DataRead[41] << 8) | DataRead[42]);
+        dtostrfd(((DataRead[43] << 8) | DataRead[44]) * 0.01f, 2, TempDataChar);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: WFreqSetPoint: %s Hz (Over Frequency drop output setpoint)"), TempDataChar);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: WFreqDroopRate: %d %% (drop output slope)"), (DataRead[45] << 8) | DataRead[46]);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: QuVupRate: %d %% (Q(U) curve up set point)"), (DataRead[47] << 8) | DataRead[48]);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: QuVlowRate: %d %% (Q(U) curve low set point)"), (DataRead[49] << 8) | DataRead[50]);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: WPowerLimitsPercent: %d %%"), (DataRead[51] << 8) | DataRead[52]);
+        dtostrfd(((DataRead[53] << 8) | DataRead[54]) * 0.01f, 2, TempDataChar);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: WWgra: %s %%"), TempDataChar);
+        dtostrfd(((DataRead[55] << 8) | DataRead[56]) * 0.1f, 1, TempDataChar);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: wWv2: %s V"), TempDataChar);
+        dtostrfd(((DataRead[57] << 8) | DataRead[58]) * 0.1f, 1, TempDataChar);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: wWv3: %s V"), TempDataChar);
+        dtostrfd(((DataRead[59] << 8) | DataRead[60]) * 0.1f, 1, TempDataChar);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: wWv4: %s V"), TempDataChar);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: wQurangeV1: %d %%"), (DataRead[61] << 8) | DataRead[62]);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: wQurangeV4: %d %%"), (DataRead[63] << 8) | DataRead[64]);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: BVoltPowerLimit: %d"), (DataRead[65] << 8) | DataRead[66]);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: WPowerManagerEnable: %d"), (DataRead[67] << 8) | DataRead[68]);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: WGlobalSeachMPPTStrartFlg: %d"), (DataRead[69] << 8) | DataRead[70]);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: WFrqProtectRestrictive: %d"), (DataRead[71] << 8) | DataRead[72]);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: WQuDelayTimer: %d sec"), (DataRead[73] << 8) | DataRead[74]);
+        AddLog(LOG_LEVEL_INFO, PSTR("SX1: WFreqActivePowerDelayTimer: %d ms"), (DataRead[75] << 8) | DataRead[76]);
+

```

Comment:
My my my am I glad this driver is an optional driver. It wastes way too much code space for non-relevant information.

You might want to dive into the use of ``%*_f`` instead of dtostrfd().